css/css-flexbox/table-as-item-stretch-cross-size-5.html

```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE html>
+<link rel="author" title="David Grogan" href="mailto:dgrogan@chromium.org">
+<link rel="help" href="https://drafts.csswg.org/css-flexbox-1/#valdef-align-items-stretch">
+<link rel="bookmark" href="https://crbug.com/1181403">
+<link rel="match" href="../reference/ref-filled-green-100px-square.xht">
+<meta name="assert" content="Calculate the caption-height correctly when flexbox stretches the table inline size to between caption's min-content and max-width, affecting the caption's block size.">
+
+<style>
+#reference-overlapped-red {
+  position: absolute;
+  background-color: red;
+  width: 100px;
+  height: 100px;
+  z-index: -1;
+}
+
+.caption-content {
+  float: left;
+  height: 10px;
+  width: 75px;
+}
+</style>
+
+<p>Test passes if there is a filled green square and <strong>no red</strong>.</p>
+<div id="reference-overlapped-red"></div>
+
+<!-- Chrome versions [84..90] incorrectly give the flexbox a height of 80px and the table a height of 100px. overflow: hidden hides the table overflow, showing the red underneath, which exposes the bug. -->
+<div style="display: flex; flex-direction: column; width: 100px; overflow: hidden;">
+  <div style="display: table; background: green; height: 80px;">
```

Comment:
@aethanyc, do you know why Firefox gives the table wrapper box a height of 30px instead of 100px? I thought the 80px here would be distributed to the table grid box, and combined with the 20px height from the caption, the final height would be 100px.

Reply:
Ok, great info and great explanation, thanks. I was worried it was a chrome bug that the test was relying on.

> Does blink reflow the table wrapper box to calculate the its flex base size?

Blink doesn't reflow the whole table wrapper box in this case (i.e. when the grid box has a preferred height in a column flexbox), it just reflows the captions and adds their resulting height to the preferred main size to get the flex item's base size. Which I think is the right thing to do.

Chrome Canary gets a [related case](https://jsfiddle.net/dgrogan/qu8ef7wn/1/) wrong: it doesn't reflow the table wrapper box to get a minimum block size below which the column flexbox cannot shrink it. I believe there are no wpt tests that exercise that bug though :(

Reply:
>  it just reflows the captions and adds their resulting height to the preferred main size to get the flex item's base size. Which I think is the right thing to do.

Sounds reasonable to me. Thank you for the ping. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1703401 for this test failure.

> Chrome Canary gets a related case wrong: it doesn't reflow the table wrapper box to get a minimum block size below which the column flexbox cannot shrink it. I believe there are no wpt tests that exercise that bug though :(

Yeah, table as flex item is definitely tricky to deal with because of table wrapper box structure  :(

Reply:
> > it just reflows the captions and adds their resulting height to the preferred main size to get the flex item's base size. Which I think is the right thing to do.
> 
> Sounds reasonable to me. Thank you for the ping. I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1703401 for this test failure.

I think I was wrong, actually. From the very last paragraph in https://drafts.csswg.org/css-flexbox/#flex-items :

> The contents of any caption boxes contribute to the calculation of the table wrapper box’s min-content and max-content sizes.

So I think the caption heights are supposed to be ignored for the flex base size. But not sure. I don't understand the next sentence that starts with "However"

Reply:
If caption heights are to be ignored for flex base size, then I think these tests are wrong; the spec calls for the table to be shorter than the height these tests call for.

css/css-flexbox/table-as-item-inflexible-in-column-2.html
css/css-flexbox/table-as-item-stretch-cross-size-3.html
css/css-flexbox/table-as-item-stretch-cross-size-5.html

Reply:
> So I think the caption heights are supposed to be ignored for the flex base size.

I don't see the spec is explicitly calling for not including caption's height in the flex base size. My interpretation to the sentence "However, like width and height, the flex longhands apply to the table box ..." is:
`flex-basis` should apply to the inner table box like `height` does, but the spec doesn't explicit mention how to calculate the flex base size of the table wrapper box.

At least, having caption height included in the flex base size make the table behave consistent in both flex and block container. https://jsfiddle.net/aethanyc/vfqe4ok2/

Do you feel it's worth filing a spec issue for clarification?

Reply:
> I don't see the spec is explicitly calling for not including caption's height in the flex base size.

Agreed, there is nothing explicit. Just seems like a weird omission in the text. Like, if including caption's height in the flex base size was intended, I feel like THAT would be explicit.

> My interpretation to the sentence "However, like width and height, the flex longhands apply to the table box ..." is:
> `flex-basis` should apply to the inner table box like `height` does, but the spec doesn't explicit mention how to calculate the flex base size of the table wrapper box.

Oh, yeah, we are saying the same thing about spec not being explicit about caption heights + flex base size, maybe I just have a different interpretation of the implications of the omission :)

> At least, having caption height included in the flex base size make the table behave consistent in both flex and block container. https://jsfiddle.net/aethanyc/vfqe4ok2/
> 
> Do you feel it's worth filing a spec issue for clarification?

Honestly, probably not. If FF and Blink both agree to add caption height to the flex base size (including for http://wpt.live/css/css-flexbox/table-as-item-stretch-cross-size-5.html where Chrome Canary and Firefox Nightly disagree today), then let's just run with that interpretation and I'll ignore my reservations about the omission we talked about above. It's probably more trouble than it's worth to get spec clarification; interop is most important and this vanishingly small use case is probably not worth our collective time to dig further into.

Does that make sense?

In any event, thanks a bunch for engaging here and adding a bunch of flex+caption tests that Chrome was failing.

Reply:
Yes, make sense to me. Let's stick with our interpretation. The current tests should guarantee the interop.

> In any event, thanks a bunch for engaging here and adding a bunch of flex+caption tests that Chrome was failing.

Thank you. It's nice to have this conversation!